src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -2,52 +2,66 @@
 
 import com.kodilla.ecommercee.domain.dto.CartDto;
 import com.kodilla.ecommercee.domain.dto.CartItemDto;
-import lombok.RequiredArgsConstructor;
+import com.kodilla.ecommercee.domain.dto.OrderDto;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-
 import java.math.BigDecimal;
+import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.List;
 
+
 @RestController
 @CrossOrigin("*")
-@RequestMapping(consumes = MediaType.APPLICATION_JSON_VALUE, value = "/carts")
+@RequestMapping("/carts")
 public class CartController {
     
-    @PostMapping(value = "/createCart/{userId}")
-    public ResponseEntity<Void> createCart(@PathVariable Long userId) {
-        List<CartItemDto> cart = new ArrayList<>();
-        //post to repository//
-        return ResponseEntity.ok().build();
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> createCart(@RequestBody CartDto cartDto) {
+        return new ResponseEntity<>("New cart was created for userid: " + cartDto.getUserId(), HttpStatus.OK);
     }
     
-    @GetMapping(value = "/{userId}/{cartId}")
-    public ResponseEntity<CartDto> getCart(@PathVariable Long userId, @PathVariable Long cartId) {
-        //get from repository//
-        //map to Dto//
-        CartDto currentCard = new CartDto(userId, cartId, new ArrayList<>());
-        return ResponseEntity.ok(currentCard);
+    @GetMapping("/products")
+    public ResponseEntity<List<CartItemDto>> getCart(@RequestParam long cartId) {
+        List<CartItemDto> cartItemDtos = new ArrayList<>();
+        cartItemDtos.add(new CartItemDto(
+                1L,"product1","description1", new BigDecimal(100),4L));
+        cartItemDtos.add(new CartItemDto(
+                2L,"product2","description2", new BigDecimal(200),3L));
+        return ResponseEntity.ok(cartItemDtos);
     }
 
-    @PostMapping(value = "{cartId}")
-    public ResponseEntity<Void> addCartItem(@PathVariable Long cartId, @RequestBody CartItemDto cartItemDto) {
-        //post item into Cart//
-        return ResponseEntity.ok().build();
+    @PutMapping("/add/{cartId}")
+    public ResponseEntity<String> addProductToCart(@PathVariable long cartId, @RequestParam long productId) {
+        // below should be executed in DbService
+        // Cart cart = findById(cartId)
+        // cart.getProductList.add(findProductById(productId);
+        // above is just rought representation of execution.
+        return new ResponseEntity<>("Product: " + productId + " was added to cart: " + cartId, HttpStatus.OK);
     }
 
-    @DeleteMapping(value = "/deleteItem/{cartItemId}")
-    public ResponseEntity<Void> removeCartItem(@PathVariable Long cartItemId,@RequestParam Long cartId) {
-        return ResponseEntity.ok().build();
+    @PutMapping("/remove{cartId}")
```

Comment:
I'm not sure about @PutMapping but OK, let it be 

 @PutMapping("/remove/{cartId}")    (should add / )

Reply:
I think it should be @DeleteMapping

Reply:
And you need CartItemId, not CartId - a user can only access their own cart so it's unnecessary to specify which one it is, instead we need to know which product to delete

Reply:
I am thinking if you're right. In database we need to know which cart is it so we can delete or add product to desired cart. If user can access only his own cart then this method should know which one it is. And how can this controller method know it without giving it cartId? Honestly I have no idea maybe you guys do have?

Reply:
This endpoint must be simple. Specification says that we should remove product from cart, so we use CartItemId and CartId. We should talk about it together, this endpoint is very problematic.

Reply:
I'm not sure if it is correct but I would do like that:
    @DeleteMapping("/{cartId}/{cartItemId}")
    public ResponseEntity<String> removeItemFromCart(@PathVariable Long cartId, @PathVariable Long cartItemId) {
        return new ResponseEntity<>("Item: " + cartItemId + " was removed from cart: " + cartId, HttpStatus.OK);
    }

Reply:
Maybe you are right